hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/HmppsPrisonerSearchExceptionHandler.kt

```diff
@@ -0,0 +1,91 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.config
+
+import jakarta.validation.ValidationException
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
+import org.springframework.http.HttpStatus.BAD_REQUEST
+import org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.AccessDeniedException
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.RestControllerAdvice
+import org.springframework.web.server.ResponseStatusException
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.exceptions.BadRequestException
+
+@RestControllerAdvice
+class HmppsPrisonerSearchIndexerExceptionHandler {
+  @ExceptionHandler(BadRequestException::class)
+  fun handleBadRequestException(e: Exception): ResponseEntity<ErrorResponse> =
+    ResponseEntity
+      .status(BAD_REQUEST)
+      .body(
+        ErrorResponse(
+          status = BAD_REQUEST,
+          userMessage = "Validation failure: ${e.message}",
+          developerMessage = e.message,
+        ),
+      ).also { log.info("Validation exception: {}", e.message) }
+
```

Comment:
The BadRequestExceptions current thrown can probably just change to validationexceptions - a later PR